### PR TITLE
Testing shared wallets

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

ADP-866

# Overview

- 73225b4fd202df2cb1026caace4a4d329d92360f
  fix e2e tests to comply with purpose=1854' for multisig keys
  
- c47591744dafb837a6492342ddc46270319d0160
  Tests for cosigner self
  
- ed846f031a2e2627155efb02037193f6cfad8525
  Tests for public keys
  
- 40698f49dc36f4e372b1afca8aeeacd311276a2f
  Tests for Account Public Keys
# Comments

### CODE REVIEW

File: test/e2e/spec/shared_spec.rb [hunk 1]
```diff
@@ -249,9 +248,246 @@
...
+        matrix_h.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(pending_wid, role, id, {hash: true})
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+      end
+
+      it "Get public key - pending wallet from acc pub key" do
+        pending "utxo_internal returns empty response"
```
Comment (piotr-iohk) [inline test/e2e/spec/shared_spec.rb:361]: @paweljakubas I've added tests covering the issue where utxo_internal returns empty response and tagged them pending for now...

File: test/e2e/spec/shared_spec.rb [hunk 2]
```diff
@@ -249,9 +248,246 @@
...
+          expect(res.to_s).to include "acct_shared_xvk"
+
+          res = SHARED.keys.create_acc_public_key(pending_wid, index, PASS, 'non_extended')
+          expect(res).to be_correct_and_respond 202
+          expect(res.to_s).to include "acct_shared_vk"
+        end
+      end
+
+      it "Create account public key - pending wallet from acc pub key" do
+        pending 'no_root key error on wallet from acc pub key'
```
Comment (piotr-iohk) [inline test/e2e/spec/shared_spec.rb:443]: Also here for the fact that currently there is 'no_root key error' returned when trying to get acc pub key from wallets created from acc pub key.

**Review by paweljakubas — APPROVED** [review]
lgtm

**Pull request merged**